notebooks/NNFullCombos.py

```diff
@@ -0,0 +1,446 @@
+    print(f"Max relative error is {np.max(rel_errors)}")
+
+    mean_errors_list.append(np.mean(errors))
+    max_errors_list.append(np.max(errors))
+    mean_rel_errors_list.append(np.mean(rel_errors))
+    max_rel_errors_list.append(np.max(rel_errors))
+
+print(AL)
+print(mean_errors_list)
+print(max_errors_list)
+print(mean_rel_errors_list)
+print(max_rel_errors_list)
+
+    # num_speed_samp = 2700000
+
+    # dt = 60 # Time step
+    # dx = 1800 # segment length
+    # # bw = np.linspace(0.135, 230.035, array_length, endpoint=True) # Trapezoidal bottom width
+    # bw = np.random.uniform(bw_min, bw_max, num_speed_samp) # Trapezoidal bottom width
+    # tw = np.random.uniform(tw_min, tw_max, num_speed_samp) # Channel top width (at bankfull)
+    # # twcc = np.linspace(0.67, 1150.17, array_length, endpoint=True) # Flood plain width
+    # # twcc = tw*  # Flood plain width tw*3
+    # n_manning = .028   # manning roughness of channel
+    # n_manning_cc = .028 # manning roughness of floodplain
+    # cs = np.random.uniform(cs_min,cs_max, num_speed_samp)# channel trapezoidal sideslope
+    # s0 = np.random.uniform(s0_min, s0_max, num_speed_samp) # Lateral inflow in this time step
+    # qup = np.random.uniform(qup_min, qup_max, num_speed_samp) # Flow from the upstream neighbor in the previous timestep
+    # # quc = np.linsprandom.uniformace(10, 1000, array_length, endpoint=True) # Flow from the upstream neighbor in the current timestep 
+    # quc = np.random.uniform(quc_min, quc_max, num_speed_samp)  # Flow from the upstream neighbor in the current timestep 
+    # # qdp = np.linspace(10, 1000, array_length, endpoint=True) # Flow at the current segment in the previous timestep
+    # qdp = np.random.uniform(qdp_min, qdp_max, num_speed_samp)  # Flow at the current segment in the previous timestep
+    # qlat = np.random.uniform(qlat_min, qlat_max, num_speed_samp) # lat inflow into current segment in the current timestep
+    # velp = .5  # Velocity in the current segment in the previous timestep NOT USED AS AN INPUT!!!
+    # depthp = np.random.uniform(depthp_min ,depthp_max , num_speed_samp) # D
+
+    # speed_x = []
+
+    # for i in range(num_samp):
+    #     speed_x.append( [normalize(qup[i],qup_max,qup_min),
+    #     normalize(quc[i],quc_max,quc_min), 
+    #     normalize(qlat[i],qlat_max,qlat_min),
+    #     normalize(qdp[i],qdp_max,qdp_min),
+    #     # dx,  
+    #     normalize(bw[i],bw_max,bw_min),
+    #     normalize(tw[i],tw_max,tw_min),
+    #     # normalize(tw[tw_o]*3,tw_max,tw_min),
+    #     # n_manning, 
+    #     # n_manning_cc, 
+    #     normalize(cs[i],cs_max,cs_min),
+    #     normalize(s0[i], s0_max, s0_min),
+    #     # velp, 
+    #     normalize(depthp[i],depthp_max,depthp_min)])
+    # speed_x = np.array(speed_x)
+
+    # import time 
+
+    # starttime=time.time()
+    # regr.predict(speed_x)
+    # print(time.time()-starttime)
+
+    # In[ ]:
+
+
+
+
```

Comment:
This needs some refactor/cleanup in general. Initial thoughts:  Try to refactor the loop mania, group all imports at the top of the file (or maybe in their own module with functions that define the operations which are imported and called from this module).  Grouping some of the code into functions makes sense, especially where I see comments of "This section does X". Make function X.